test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +605,141 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
 		})
 	})
+
+	ginkgo.When("Reconciling clusterQueue pending workload status", func() {
```

Comment:
I would like to have also a test for the configuration when the feature is disabled. However, but we can defer it to a follow up.

Reply:
Maybe, this is indeed a little bit of work. However, I would like to have a tested way out for users for who the feature is breaking (if any). OTOH, the feature enabled on low scale is unlikely to break any users. So, indeed, it is a judgement call. But then, maybe we don't need the extra `if` in start?

Anyway, I was thinking we could have an extra `Describe` in test, where we setup with `MaxCount=0`. The test would create a workload and assert the head is nil. Let's see what @alculquicondor thinks. I'm fine to call it overkill as well.

Reply:
After a second thought I suggest to add unit tests (or expand the existing) with that scenario. Unit tests will be valuable as then we can set `UpdateStatusInterval` to some small value (like 10ms), to make the testing loop fast.